tiledb/sm/array_schema/array_schema.h

```diff
@@ -249,6 +249,34 @@ class ArraySchema {
    */
   Status drop_attribute(const std::string& attr_name);
 
+  /**
+   * Renames an attribute.
+   *
+   * @param old_name The old attribute name
+   * @param new_name The new attribute name
+   * @param is_new_uri The flag to generate new array schema uri
+   * @return Status
+   */
+  Status rename_attribute(
+      const std::string& old_name,
+      const std::string& new_name,
+      bool is_new_uri);
+
+  /**
+   * Returns a vector of used names of attributes
+   *
+   * @return all used names
+   */
+  std::vector<std::string> attribute_used_names() const;
```

Comment:
I suggest renaming to `attribute_old_names`.

Reply:
renamed to attribute_old_names